File: Assignment2/VB.NET/AritMeanDistr/AritMeanDistr/Form1.vb

Imports AritMeanDistr.WindowsFormsApp1

Public Class Form1
    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load


    End Sub

    Private Sub RichTextBox2_TextChanged(sender As Object, e As EventArgs) Handles RichTextBox2.TextChanged

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Timer1.Interval = 2
        Timer1.Start()
        Button3.Visible = True
        Button2.Visible = True
        Button1.Visible = False
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Timer2.Interval = 10
        Timer2.Start()
        Button2.Visible = False
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Timer1.[Stop]()
        Timer2.[Stop]()
        Button1.Visible = True
        Button2.Visible = True
    End Sub

    Private r As Random = New Random()
    Private mc As MediaCalOnline = New MediaCalOnline()
    Private dc As Distribuzione = New Distribuzione()


    Private Sub Timer1_Tick_1(sender As Object, e As EventArgs) Handles Timer1.Tick
        Dim height As Double
        Dim studente As ElementoDisribuzione = New ElementoDisribuzione("Student")
        height = r.NextDouble() + r.Next(140, 200)
        studente.setVariable("height", height)
        mc.addElemento(studente)
        dc.addElementoDef(studente)
        RichTextBox1.Text += "nuovo Studente altezza: " & height & Environment.NewLine
    End Sub

    Private Sub Timer2_Tick_1(sender As Object, e As EventArgs) Handles Timer2.Tick
        Dim i As Double
        Dim distTest As SortedDictionary(Of Tuple(Of Double, Double), Integer) = New SortedDictionary(Of Tuple(Of Double, Double), Integer)()
        mc.getMedia("height", i)

        If dc.getdistribuzione("height", distTest) Then
            RichTextBox3.Text = ""

            For Each item In distTest
                RichTextBox3.Text += item.Key.Item1 & "-" & item.Key.Item2 & " presenta " & item.Value & " entita " & Environment.NewLine
            Next
        End If

        RichTextBox2.Text = "the A M of the height is" & i & Environment.NewLine
    End Sub
End Class

File: Assignment5/C#/OnlineArithmeticMean/WindowsFormsApp1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WindowsFormsApp1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        Random r = new Random();
        MediaCalOnline mc=new MediaCalOnline();
        Distribuzione dc = new Distribuzione();
        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void button1_Click(object sender, EventArgs e)
        {
            timer1.Interval = 2;
            timer1.Start();
            button3.Visible = true;
            button2.Visible = true;
            button1.Visible = false;
        }

        private void button2_Click(object sender, EventArgs e)
        {
            timer2.Interval = 10;
            timer2.Start();
            button2.Visible = false;
            
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            double height;
            ElementoDisribuzione studente = new ElementoDisribuzione("Student");
            height =r.NextDouble()+r.Next(140,200);
            studente.setVariable("height", height);
            mc.addElemento(studente);
            dc.addElementoDef(studente);

            richTextBox1.Text += "nuovo Studente altezza: " + height + Environment.NewLine;




        }

        private void timer2_Tick(object sender, EventArgs e)
        {
            double i;
            SortedDictionary<Tuple<double, double>, int> distTest =new  SortedDictionary<Tuple<double, double>, int>();
            mc.getMedia("height", out i);
            if (dc.getdistribuzione("height", out distTest))
            {
                richTextBox3.Text = "";
                foreach (var item in distTest)
                {
                    richTextBox3.Text += item.Key.Item1 + "-" + item.Key.Item2 + " presenta " + item.Value + " entita "+Environment.NewLine;

                }
            }

            richTextBox2.Text = "the A M of the height is" + i + Environment.NewLine;
        }

        private void button3_Click(object sender, EventArgs e)
        {
            timer1.Stop();
            timer2.Stop();
            button1.Visible = true;
            button2.Visible = true;
        }

      
    }
}
